plugin.video.francetv/resources/lib/api.py

```diff
@@ -0,0 +1,512 @@
+
+        if program:
+            info["tvshowtitle"] = program
+
+        if not info["year"] and item.get("broadcast_begin_date"):
+            info["aired"] = time.strftime(
+                "%Y-%m-%d",  # type: ignore
+                time.localtime(item["broadcast_begin_date"]),
+            )
+
+        if item.get("begin_date"):
+            info["dateadded"] = time.strftime(
+                "%Y-%m-%d %H:%M:%S",  # type: ignore
+                time.localtime(item["begin_date"]),
+            )
+
+        if (
+            info.get("tvshowtitle")
+            or info.get("episode")
+            or info.get("season")
+        ):
+            info["mediatype"] = "episode"
+        else:
+            info["mediatype"] = "movie"
+
+        properties["isPlayable"] = "true"
+
+        return result
+
+    def get_collection(self, path, level=None):
+        # type: (Text, Optional[int]) -> Generator[ParsedItem, None, None]
+
+        data = self._query_api(path)
+
+        if isinstance(data, dict):
+            cursor = data.get("cursor")
+            parent_item = data.get("item") or data
+            collection = data.get("collections") or data.get("items") or []
+        else:
+            cursor = None
+            parent_item = {}
+            collection = data
+
+        if level is not None and level < len(collection):
+            cursor = None
+            parent_item = collection[level]
+            collection = collection[level].get("items") or []
+
+        # Sub-category items only provides incomplete list of programs and
+        # videos, use our own extra items insteads
+        if parent_item.get("type") != "sous_categorie":
+            for index, item in enumerate(collection):
+                parsed_item = self._parse_item(item, path, index, parent_item)
+                if parsed_item:
+                    yield parsed_item
+
+        if path == "generic/channels":
+            # Add "virtual" Okoo channel in the channel collection, as done on
+            # the france.tv website
+            yield ParsedItem(
+                "Okoo",
+                {"mode": "collection", "path": "apps/categories/enfants"},
+                {"plot": "Okoo"},
+                {"icon": _CHANNEL_ICONS["okoo"]},
+                {},
+            )
+
+        # Extra items
+
+        parent_item_type = parent_item.get("type")
+
+        if parent_item_type in [
+            "categorie",
+            "channel",
+            "program",
+            "region",
+            "sous_categorie",
+        ]:
+            collection_id = (
+                parent_item.get("channel_path")
+                or parent_item.get("program_path")
+                or parent_item.get("region_path")
+                or parent_item.get("url_complete")
+            )
+
+            # Add "all TV shows"/"all programs" item
+            if parent_item_type != "program":
+                yield ParsedItem(
+                    "Tous les programmes",
```

Comment:
I guess this is shown in the ui? If so, please localize

Reply:
Fixed